packages/nimble-components/src/rich-text/models/markdown-parser.ts

```diff
@@ -53,7 +64,69 @@ export class RichTextMarkdownParser {
             'newline'
         ]);
 
-        supportedTokenizerRules.validateLink = href => /^https?:\/\//i.test(href);
+        // MarkdownIt use: https://markdown-it.github.io/markdown-it/#MarkdownIt.use
+        supportedTokenizerRules.use(
+            markdown => {
+                // `mention` rule is added before `autolink` rule to get the highest preference when getting the DOM output
+                // MarkdownIt before: https://markdown-it.github.io/markdown-it/#Ruler.before
+                markdown.inline.ruler.before(
+                    'autolink',
+                    'mention',
+                    (state, _silent) => {
+                        const mentionHref = this.getMentionHref(state);
+
+                        if (!mentionHref || !mentionsMap?.size) {
+                            return false;
+                        }
+
+                        const currentMention = Array.from(
+                            mentionsMap.values()
+                        ).find(
+                            mention => mention.validConfiguration
+                                && this.validateMentionPattern(
+                                    mention,
+                                    mentionHref
+                                )
+                        );
```

Comment:
Wait... what's the point of creating the mentionsMap if you just turn it into an array?? It's doesn't look like anything tried to lookup stuff in the map at all lol.

Also wouldn't this code run frequently? All this validation and stuff should have happened way before parseMarkdownToDOM is called and ideally only once.

What the updateMentionsMap function needs to do is capture all the stuff needed for the markdown parser to run at that time. Check validity, create regex objects, etc. Create a configuration that markdown-parser can use without coupling to all these other parts of the system.

Reply:
> what's the point of creating the mentionsMap if you just turn it into an array?

I have updated it to an array and identified what type of mention string is in the markdown input in markdown-parser.ts file.

As discussed offline over a call, we aligned to have a separate model class with only the necessary details of what the markdown parser requires. Also checked the validation at the time of creating the parser configuration instead of handling it within the parser. This way only when the mention internals change, all the regex pattern objects are created and the parser just gets the necessary details like the display name when the mention href in the markdown maps to.

Reply:
@rajsite I have updated the parser configuration as per our discussion last week and I hope is current implementation is what you would want us to make. If you have any concerns with the change, we will either update it in the next follow-up PRs for @mention or create a new PR to address your feedback. For now, I am closing this comment.